src/libopenrave/kinbodygrab.cpp

```diff
@@ -674,15 +709,27 @@ void KinBody::GrabbedInfo::DeserializeJSON(const rapidjson::Value& value, dReal
         _trelative.trans *= fUnitScale;
     }
     orjson::LoadJsonValueByKey(value, "ignoreRobotLinkNames", _setIgnoreRobotLinkNames);
+    if( value.HasMember("userData" ) ) {
+        _prUserData.reset(new rapidjson::Document());
+        orjson::SaveJsonValue(*_prUserData, value["userData"], _prUserData->GetAllocator());
+    }
+    else {
+        _prUserData.reset();
+    }
 }
 
-void KinBody::GrabbedInfo::serialize(std::ostream& o) const
+void KinBody::GrabbedInfo::serialize(std::ostream& os) const
 {
-    o << _grabbedname << " ";
-    o << _robotlinkname << " ";
-    SerializeRound(o, _trelative);
+    os << _grabbedname << " ";
+    os << _robotlinkname << " ";
+    SerializeRound(os, _trelative);
     for( std::set<std::string>::const_iterator it = _setIgnoreRobotLinkNames.begin(); it != _setIgnoreRobotLinkNames.end(); ++it ) {
-        o << (*it) << " ";
+        os << (*it) << " ";
+    }
+    if( !!_prUserData ) {
+        // using 'void DumpJson(Value, ostream, unsigned int)' so that std::setprecision is respected
```

Comment:
minor, but ``DumpJson`` (rapidjson's ``OStreamWrapper``) currently doesn't respect properties like precision set on ostream. Using ``ostream`` version overload is for performance and for potential future enhancement to respect such property.

Reply:
sure, I understand rapidjson handles the ostream. fixed comment.